vdev/src/testing/integration.rs

```diff
@@ -129,59 +131,72 @@ impl IntegrationTest {
     }
 
     pub fn start(&self) -> Result<()> {
-        self.runner.ensure_network()?;
+        if let Some(compose_path) = &self.compose_path {
+            self.runner.ensure_network()?;
 
-        if self.envs_dir.check_active(&self.environment)? {
-            bail!("environment is already up");
-        }
+            if self.envs_dir.check_active(&self.environment)? {
+                bail!("environment is already up");
+            }
 
-        self.prepare_compose()?;
-        self.run_compose("Starting", &["up", "--detach"], &self.env_config)?;
+            #[cfg(unix)]
+            unix::prepare_compose_volumes(compose_path, &self.test_dir)?;
+            self.run_compose("Starting", &["up", "--detach"], &self.env_config)?;
 
-        self.envs_dir.save(&self.environment, &self.env_config)
+            self.envs_dir.save(&self.environment, &self.env_config)
+        } else {
```

Comment:
Worth a short comment to explain what scenario this applies to ? (essentially your opening comment from the PR)

Or even, instead of checking the compose path in both start and stop, adding a helper function that returns a bool with a descriptive function name. In it, it just does the check for self.compose_path.is_some().